### PR TITLE
Lowercase and trim processors

### LINKED ISSUE #697: String Manipulation Processor
Data Prepper should have a Processor for string manipulation.

Some candidate operations:

* `uppercase` - Make a string all uppercase
* `lowercase` - Make a string all lowercase
* `trim` - Trim whitespace from the start and end of a string
* `split` - Split a string on a delimiter into an array of strings
* `join` - Create a new string from other fields, separated by a delimiter
* `format` - Create a new string from a format string and other values in an event

This should be a new Processor. It can also deprecate the existing `string_converter` plugin. Unlike the existing `string_converter` processor, this new processor will rely on updating keys. See https://github.com/opensearch-project/data-prepper/pull/753/files#r770751733 for more information on the difference between the plugins.

### PR DESCRIPTION
**Pull request by dapowers87**
### Description
This adds the `LowercaseStringProcessor` and `TrimStringProcessor` and its associated unit tests
 
### Issues Resolved
#697 
 
### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: data-prepper-plugins/mutate-string-processors/src/main/java/com/amazon/dataprepper/plugins/processor/mutatestring/LowercaseStringProcessor.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+package com.amazon.dataprepper.plugins.processor.mutatestring;
+
+import com.amazon.dataprepper.metrics.PluginMetrics;
+import com.amazon.dataprepper.model.annotations.DataPrepperPlugin;
+import com.amazon.dataprepper.model.annotations.DataPrepperPluginConstructor;
+import com.amazon.dataprepper.model.event.Event;
+import com.amazon.dataprepper.model.processor.Processor;
+import java.util.Locale;
+
+/**
```
Comment (graytaylor0) [inline data-prepper-plugins/mutate-string-processors/src/main/java/com/amazon/dataprepper/plugins/processor/mutatestring/LowercaseStringProcessor.java:15]: If we need comments at the top of a processor, we should make then JavaDocs

File: data-prepper-plugins/mutate-string-processors/src/main/java/com/amazon/dataprepper/plugins/processor/mutatestring/LowercaseStringProcessor.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    @Override
+    protected void performKeyAction(Event recordEvent, String key, String value)
+    {
+        recordEvent.put(key, value.toLowerCase(Locale.ROOT));
+    }
+
+    @Override
+    protected String getKey(String entry) {
+        return entry;
+    }
```
Comment (graytaylor0) [inline data-prepper-plugins/mutate-string-processors/src/main/java/com/amazon/dataprepper/plugins/processor/mutatestring/LowercaseStringProcessor.java:35]: This is more of a question about the abstract class but I'm not sure I understand the point of this function. Which String processors will not just return the String they are provided?

File: data-prepper-plugins/mutate-string-processors/src/test/java/com/amazon/dataprepper/plugins/processor/mutatestring/TrimStringProcessorTests.java [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+    @Mock
+    private StringProcessorConfig<String> config;
+
+    @BeforeEach
+    public void setup() {
+        lenient().when(config.getIterativeConfig()).thenReturn(Collections.singletonList("message"));
+    }
+
+    @Test
```
Comment (graytaylor0) [inline data-prepper-plugins/mutate-string-processors/src/test/java/com/amazon/dataprepper/plugins/processor/mutatestring/TrimStringProcessorTests.java:43]: Let's add a test case where there is whitespace in the middle somewhere (" thisis amessage "). You could make this test a `@ParameterizedTest` as well to make it cleaner.

File: data-prepper-plugins/mutate-string-processors/src/main/java/com/amazon/dataprepper/plugins/processor/mutatestring/LowercaseStringProcessor.java (hunk 1 above)
Comment (dapowers87) [reply data-prepper-plugins/mutate-string-processors/src/main/java/com/amazon/dataprepper/plugins/processor/mutatestring/LowercaseStringProcessor.java:15]: That was my intent. Is this not javadocs?

File: data-prepper-plugins/mutate-string-processors/src/main/java/com/amazon/dataprepper/plugins/processor/mutatestring/LowercaseStringProcessor.java (hunk 2 above)
Comment (dapowers87) [reply data-prepper-plugins/mutate-string-processors/src/main/java/com/amazon/dataprepper/plugins/processor/mutatestring/LowercaseStringProcessor.java:35]: If the processor has a sub Entry class, then getKey would pass in the Entry object and you would return the appropriate value.

Comment (cmanning09) [reply data-prepper-plugins/mutate-string-processors/src/main/java/com/amazon/dataprepper/plugins/processor/mutatestring/LowercaseStringProcessor.java:35]: That would be `substitute`, `split`, `replace`, & `update`. Am I missing any @dapowers87?

Comment (dapowers87) [reply data-prepper-plugins/mutate-string-processors/src/main/java/com/amazon/dataprepper/plugins/processor/mutatestring/LowercaseStringProcessor.java:35]: `convert` as well

File: data-prepper-plugins/mutate-string-processors/src/main/java/com/amazon/dataprepper/plugins/processor/mutatestring/LowercaseStringProcessor.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+ */
+@DataPrepperPlugin(name = "lowercase_string", pluginType = Processor.class, pluginConfigurationType = StringProcessorConfig.class)
+public class LowercaseStringProcessor extends AbstractStringProcessor<String> {
+    @DataPrepperPluginConstructor
+    public LowercaseStringProcessor(final PluginMetrics pluginMetrics, final StringProcessorConfig<String> config) {
+        super(pluginMetrics, config);
+    }
+
+    @Override
+    protected void performKeyAction(Event recordEvent, String key, String value)
```
Comment (cmanning09) [inline data-prepper-plugins/mutate-string-processors/src/main/java/com/amazon/dataprepper/plugins/processor/mutatestring/LowercaseStringProcessor.java:27]: Let's mark the parameters as final to prevent re-assignment in both `performKeyAction` and `getKey` for both new processors

**Review by cmanning09 — DISMISSED** [review]
I have one minor comment, otherwise looks good.

File: data-prepper-plugins/mutate-string-processors/src/main/java/com/amazon/dataprepper/plugins/processor/mutatestring/LowercaseStringProcessor.java (hunk 1 above)
Comment (graytaylor0) [reply data-prepper-plugins/mutate-string-processors/src/main/java/com/amazon/dataprepper/plugins/processor/mutatestring/LowercaseStringProcessor.java:15]: Yeah they are my bad. We could add the `@since 1.3` tag here but not a big deal.

**Review by graytaylor0 — APPROVED** [review]
Looks good!

**Review by cmanning09 — APPROVED** [review]
